examples/components/NVFLARE/client/run.py

```diff
@@ -0,0 +1,202 @@
+        type=str,
+        required=False,
+        default="CLIENT_DATA_PATH",
+        help="the name of the env variable to set with the mount path of the client_data folder",
+    )
+
+    return parser
+
+
+def run_cli_command(cli_command: list, timeout: int = None, custom_env: dict = None):
+    """Runs subprocess for a cli setup command"""
+    logger = logging.getLogger()
+    logger.info(f"Launching cli with command: {cli_command}")
+    cli_command_call = subprocess.run(
+        cli_command,
+        # stdout=PIPE,
+        # stderr=PIPE,
+        universal_newlines=True,
+        check=False,  # will not raise an exception if subprocess fails
+        timeout=timeout,  # TODO: more than a minute would be weird?
+        env=custom_env,
+    )
+    logger.info(f"return code: {cli_command_call.returncode}")
+
+    if cli_command_call.returncode != 0:
+        raise RuntimeError("Cli command returned code != 0")
+
+    return cli_command_call.returncode
+
+
+def fetch_server_ip(
+    mlflow_run,
+    federation_identifier,
+    fetch_name=False,
+    target_run_tag="mlflow.rootRunId",
+    timeout=600,
+):
+    """Fetches the server ip and name from the mlflow run tags.
+
+    Args:
+        mlflow_run (mlflow.entities.Run): the mlflow run to fetch the tags from
+        federation_identifier (str): the federation identifier to use to find the server ip
+        fetch_name (bool, optional): whether to fetch the server name. Defaults to False.
+        target_run_tag (str, optional): the tag to use to find the root run. Defaults to "mlflow.rootRunId".
+        timeout (int, optional): the timeout in seconds to wait for the server ip. Defaults to 600.
+
+    Returns:
+        str, str: the server ip and name
+    """
+    logger = logging.getLogger(__name__)
+
+    mlflow_client = mlflow.tracking.client.MlflowClient()
+    logger.info(f"run tags: {mlflow_run.data.tags}")
+    logger.info(
+        f"target tag {target_run_tag}={mlflow_run.data.tags.get(target_run_tag)}"
+    )
+    target_run_id = mlflow_run.data.tags.get(target_run_tag)
+
+    server_ip = None
+    server_name = None
+
+    fetch_start_time = time.time()
+
+    while (fetch_name and server_name is None) or server_ip is None:
+        logger.info(f"Checking out tag server_ip...")
+        mlflow_root_run = mlflow_client.get_run(target_run_id)
+
+        server_ip_tag = f"{federation_identifier}.server_ip"
+
+        if server_ip_tag in mlflow_root_run.data.tags:
+            server_ip = mlflow_root_run.data.tags[server_ip_tag]
+            logger.info(f"server_ip found: {server_ip}")
+
+        if fetch_name:
+            server_name_tag = f"{federation_identifier}.server_name"
+
+            if server_name_tag in mlflow_root_run.data.tags:
+                server_name = mlflow_root_run.data.tags[server_name_tag]
+                logger.info(f"server_name found: {server_name}")
+
+        if ((fetch_name and server_name is None) or server_ip is None) and (
```

Comment:
The idea was to avoid burning compute in region A while region B was being provisioned, if that provisioning was blocked for way too long. The timeout is short (10 mins), but could be extended. Or we could figure out another strategy later after this PR is merged.